opentelemetry-resource-detectors/src/host.rs

```diff
@@ -4,30 +4,41 @@
 use opentelemetry::KeyValue;
 use opentelemetry_sdk::resource::ResourceDetector;
 use opentelemetry_sdk::Resource;
+use std::env::consts::ARCH;
 use std::fs::read_to_string;
 use std::path::Path;
 use std::time::Duration;
 
-/// Detect the unique host ID.
+/// Detect host information.
 ///
-/// This detector looks up the host id using the sources defined
-/// in the OpenTelemetry semantic conventions [`host.id from non-containerized systems`].
+/// This resource detector returns the following information:
 ///
-/// [`host.id from non-containerized systems`]: https://opentelemetry.io/docs/specs/semconv/resource/host/#collecting-hostid-from-non-containerized-systems
+/// - [`host.id from non-containerized systems`]: https://opentelemetry.io/docs/specs/semconv/resource/host/#collecting-hostid-from-non-containerized-systems
+/// - Host architecture (host.arch).
 pub struct HostResourceDetector {
     host_id_detect: fn() -> Option<String>,
 }
```

Comment:
I understand it is not related to this PR, but using a function pointer is bit unconventional for Rust.  Closure or trait object could be more idiomatic. Something like (not tested):

```rust
pub struct HostResourceDetector {
    host_id_detect: Box<dyn HostIdDetector>,
}

trait HostIdDetector {
    fn detect(&self) -> Option<String>;
}
#[cfg(target_os = "linux")]
struct LinuxHostIdDetector;

#[cfg(target_os = "linux")]
impl HostIdDetector for LinuxHostIdDetector {
    fn detect(&self) -> Option<String> {
       Some("my_host_id")
   }
}
#[cfg(not(target_os = "linux"))]
struct NonLinuxHostIdDetector;

#[cfg(not(target_os = "linux"))]
impl HostIdDetector for NonLinuxHostIdDetector {
    fn detect(&self) -> Option<String> {
        None
    }
}
```
But not for this PR. Commented just that we know it can be improved :)